test/e2e/framework/networking_utils.go

```diff
@@ -255,10 +256,15 @@ func (config *NetworkingTestConfig) GetEndpointsFromContainer(protocol, containe
 					eps.Insert(trimmed)
 				}
 			}
-			return eps, nil
+			// Return immediately when we successfully fetch endpoints
+			if len(eps) > 0 {
```

Comment:
Could be great if we document what exactly `GetEndpointsFromContainer()`  is for. This use case seem to be tricky.

Reply:
It's fixed now. PTAL.